File: tests/rustdoc-js/where-clause.rs
pub struct Nested;

pub trait Trait<T> {
    fn thank_you(x: T);
}

pub fn abracadabra<X>(_: X) where X: Trait<Nested> {}

pub fn alacazam<X>() -> X where X: Trait<Nested> {}

pub trait T1 {}
pub trait T2<'a, T> {
    fn please(_: &'a T);
}

pub fn presto<A, B>(_: A, _: B) where A: T1, B: for <'b> T2<'b, Nested> {}

pub trait Shazam {}

pub fn bippety<X>() -> &'static X where X: Shazam {
    panic!()
}

pub struct Drizzel<T>(T);

impl<T> Drizzel<T> {
    pub fn boppety(&self) -> &T where T: Shazam {
        panic!();
    }
}

File: tests/rustdoc-js/where-clause.js
const EXPECTED = [
    {
        'query': 'trait<nested>',
        'in_args': [
           { 'path': 'where_clause', 'name': 'abracadabra' },
        ],
    },
    {
        'query': '-> trait<nested>',
        'others': [
            { 'path': 'where_clause', 'name': 'alacazam' },
        ],
    },
    {
        'query': 't1, t2',
        'others': [
            { 'path': 'where_clause', 'name': 'presto' },
        ],
    },
    {
        'query': '-> shazam',
        'others': [
            { 'path': 'where_clause', 'name': 'bippety' },
            { 'path': 'where_clause::Drizzel', 'name': 'boppety' },
        ],
    },
    {
        'query': 'drizzel -> shazam',
        'others': [
            { 'path': 'where_clause::Drizzel', 'name': 'boppety' },
        ],
    },
];
